Add docs for themes extension type (#20561)
directus/directus: docs/extensions/themes.md

});
```

### Available Options

- `name` - The displayed name for this theme. This must be unique within your Directus project.

> - `appearance` - The section where this theme is displayed. Options are `'light'` or `'dark'`.

Reviewer: I would put it more along these lines:
```suggestion
- `appearance` - To which appearance mode the theme belongs to, `light` or `dark`.
```

- `rules` - A set of theming rules from the theme schema.

### Available Rules

Rules that are configured in the `rules` property have to adhere to the properties defined in the Rules section of the
[theme schema](https://github.com/directus/directus/blob/main/packages/themes/src/schemas/theme.ts). This includes
matching how properties are nested in the JavaScript object, for example:

```js{3-5}
rules: {
	borderRadius: '24px',
	navigation: {
		background: 'rebeccapurple'
	}
}
```

Any rules that are not defined will fallback to the default theme for it's appearance. See the
([`default dark theme`](https://github.com/directus/directus/blob/main/packages/themes/src/themes/dark/default.ts) and
[`default light theme`](https://github.com/directus/directus/blob/main/packages/themes/src/themes/dark/default.ts)).

> We recommend using TypeScript for this extension type. The `defineTheme` function is typed to properly auto-complete all

Reviewer: ```suggestion
We recommend using TypeScript for this extension type. The `defineTheme` function is typed to properly check and auto-complete all
```
